crates/sui-types/src/funds_accumulator.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::base_types::SuiAddress;
+use crate::SUI_FRAMEWORK_ADDRESS;
+use move_core_types::annotated_value::{MoveFieldLayout, MoveStructLayout, MoveTypeLayout};
+use move_core_types::ident_str;
+use move_core_types::identifier::IdentStr;
+use move_core_types::language_storage::{StructTag, TypeTag};
+use move_core_types::u256::U256;
+use serde::Deserialize;
+use serde::Serialize;
+
+pub const FUNDS_ACCUMULATOR_MODULE_NAME: &IdentStr = ident_str!("funds_accumulator");
+pub const WITHDRAWAL_STRUCT_NAME: &IdentStr = ident_str!("Withdrawal");
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct Withdrawal {
+    pub owner: SuiAddress,
+    pub limit: U256,
```

Comment:
Does this need to be optional in order to support reserver all down the road?

Reply:
I see. Could you add a comment somewhere too?
Maybe obvious based on the definition but it implies that if a transaction reserves all balance, it cannot be executed until the immediate previous settlement finishes (so that it knows the exact balance at that point).